tests/drift_kinetic_push/drift_kinetic_grid_ex2.cpp

```diff
@@ -0,0 +1,342 @@
+    "Gradient B field mismatch. Analytical: (%.8e %.8e %.8e), Grid: (%.8e %.8e %.8e), Error: %.8e",
+    REP3_A(gradB_analytical), REP3_A(gradB_grid), error_gradB);
+
+  PetscFunctionReturn(PETSC_SUCCESS);
+}
+
+PetscErrorCode print_statistics(const TestStatistics& stats)
+{
+  PetscFunctionBeginUser;
+
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "\n=== TEST STATISTICS ===\n"));
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "Simulation time: %.6e\n", stats.simulation_time));
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "Total steps: %d\n", stats.total_steps));
+
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "\nField comparison errors:\n"));
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "  Max E field error:     %.8e\n", stats.max_field_error_E));
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "  Max B field error:     %.8e\n", stats.max_field_error_B));
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "  Max gradB field error: %.8e\n", stats.max_field_error_gradB));
+
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "\nTrajectory comparison:\n"));
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "  Max position error:    %.8e\n", stats.max_position_error));
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "  Final pos analytical:  (%.6e %.6e %.6e)\n",
+           REP3_A(stats.final_position_analytical)));
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "  Final pos grid:        (%.6e %.6e %.6e)\n",
+           REP3_A(stats.final_position_grid)));
+
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "\nIteration statistics:\n"));
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "  Total iterations (analytical): %d\n", stats.total_iterations_analytical));
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "  Total iterations (grid):       %d\n", stats.total_iterations_grid));
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "  Avg iterations (analytical):   %.2f\n",
+           (PetscReal)stats.total_iterations_analytical / stats.total_steps));
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "  Avg iterations (grid):         %.2f\n",
+           (PetscReal)stats.total_iterations_grid / stats.total_steps));
+
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "\nConstraint violations:\n"));
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "  Boundary violations:    %d\n", stats.boundary_violations));
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "  Cell size violations:   %d\n", stats.cell_size_violations));
+  PetscCall(PetscPrintf(PETSC_COMM_WORLD, "  Max displacement/step:  %.8e\n", stats.max_displacement_per_step));
+
+  PetscFunctionReturn(PETSC_SUCCESS);
+}
+
+int main(int argc, char** argv)
+{
+  PetscFunctionBeginUser;
+  PetscCall(PetscInitialize(&argc, &argv, nullptr, help));
+
+  PetscReal omega_dt;
+  PetscCall(get_omega_dt(omega_dt));
+
+  dt = omega_dt / B0.length();
+  geom_nt = 100;
+  diagnose_period = geom_nt / 1;
+  geom_nx = 20;
+  geom_ny = 5;
+  geom_nz = 100;
+  dx = 1.;
+  dy = 1.;
+  dz = 1.;
+
+
+  World::set_geometry(
+    geom_nx, geom_ny, geom_nz, geom_nt,
+    dx, dy, dz, dt,
+    diagnose_period);
```

Comment:
`World::set_geometry()` updates global constants:

choose either

```suggestion
  World::set_geometry(20, 5, 100, 100, 1.0, 1.0, 1.0, omega_dt / B0.length(), 100);
```

or remove the duplicating logic of `World::set_geometry()`